### DISCUSSION TITLE
Figuring out a configuration for PVE Perl Style

### DISCUSSION THREAD

**Discussion by cheese** [perltidy/perltidy#95] [Q&A] [answered]
Proxmox VE is a long-standing open source qemu/kvm/lxc management stack written in Perl. The project deploys a legacy [Perl Style](https://pve.proxmox.com/wiki/Perl_Style_Guide). I am trying to figure out a Perltidy configuration for this style.

I read through the manpage of perltidy. The current status of my perltidyrc is

```
--indent-columns=4
--entab-leading-whitespace=8
--maximum-line-length=100
--weld-nested-containers
--paren-tightness=2
--square-bracket-tightness=0
--brace-tightness=0
--cuddled-else
--opening-token-right
--opening-brace-always-on-right
--stack-all-containers
--stack-closing-block-brace
--tight-secret-operators

--no-outdent-long-lines
--continuation-indentation=4
--novalign
--noblanks-before-comments
--character-encoding=utf8
--warning-output
--output-line-ending=unix
--space-for-semicolon
--minimum-space-to-comment=1
--ignore-side-comment-lengths
```

But there are still several issues that I didn't find solutions.

The first issue is about [wrapping arugments](https://pve.proxmox.com/wiki/Perl_Style_Guide#Wrapping_Arguments). Is it supported by a combination of current options?

Unformatted code:
```
my $result = PVE::Some::Long::Module::Name::get_foo_bar_baz($argument_one, $argument_two, $argument_three, $argument_last);
```

What I got from perltidy:
```
my $result =
    PVE::Some::Long::Module::Name::get_foo_bar_baz($argument_one, $argument_two, $argument_three,
    $argument_last);
```

What expected:
```
my $result = PVE::Some::Long::Module::Name::get_foo_bar_baz(
    $argument_one,
    $argument_two,
    $argument_three,
    $argument_last,
);
```

NB:
1. Line breaks after the start of brace.
2. Each argument in its own line
3. Trailing comma added after the last argument
4. `);` at its own line with no indent.

**Comment by shancock9** [comment, marked as answer]
Try adding the parameter --mft=1 (--maximum-fields-per-table=1) to see if it solves the problem. It gives your expected result in this particular case.


Regarding wrapping, there is a brief discussion of when it is turned on in

https://github.com/perltidy/perltidy/issues/78

and a tentative parameter (-bopl=n) was added to control it. But the parameter is not currently activated (but could be).

**Reply by shancock9** [reply]
Regarding the trailing comma issue, perltidy will retain existing trailing commas but does not have an option to automatically add them.
